src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -891,19 +894,72 @@ namespace winrt::TerminalApp::implementation
     // Arguments:
     // - sender: the control that originated this event (TabViewItem)
     // - eventArgs: the event's constituent arguments
-    void TerminalPage::_OnTabClick(const IInspectable& sender, const Windows::UI::Xaml::Input::PointerRoutedEventArgs& eventArgs)
+    void TerminalPage::_OnPointerPressed(const IInspectable& sender, const Windows::UI::Xaml::Input::PointerRoutedEventArgs& eventArgs)
     {
         if (eventArgs.GetCurrentPoint(*this).Properties().IsMiddleButtonPressed())
         {
-            const auto tabViewItem = sender.try_as<MUX::Controls::TabViewItem>();
-            if (auto tab{ _GetTabByTabViewItem(tabViewItem) })
+            if (const auto tabViewItem{ sender.try_as<MUX::Controls::TabViewItem>() })
             {
-                _HandleCloseTabRequested(tab);
+                _pointerMiddleButtonPressed = tabViewItem.CapturePointer(eventArgs.Pointer());
+                _pointerMiddleButtonExited = false;
+            }
+            eventArgs.Handled(true);
+        }
+    }
+
+    // Method Description:
+    // - Tracking pointer state for tab remove
+    // Arguments:
+    // - sender: the control that originated this event (TabViewItem)
+    // - eventArgs: the event's constituent arguments
+    void TerminalPage::_OnPointerReleased(const IInspectable& sender, const Windows::UI::Xaml::Input::PointerRoutedEventArgs& eventArgs)
+    {
+        if (_pointerMiddleButtonPressed && !eventArgs.GetCurrentPoint(*this).Properties().IsMiddleButtonPressed())
+        {
+            _pointerMiddleButtonPressed = false;
+            if (const auto tabViewItem{ sender.try_as<MUX::Controls::TabViewItem>() })
+            {
+                tabViewItem.ReleasePointerCapture(eventArgs.Pointer());
+                auto tab = _GetTabByTabViewItem(tabViewItem);
+                if (!_pointerMiddleButtonExited && tab)
+                {
+                    _HandleCloseTabRequested(tab);
+                }
+            }
+            eventArgs.Handled(true);
+        }
+    }
+
+    // Method Description:
+    // - Tracking pointer state for tab remove
+    // Arguments:
+    // - sender: the control that originated this event (TabViewItem)
+    // - eventArgs: the event's constituent arguments
+    void TerminalPage::_OnPointerEntered(const IInspectable& sender, const Windows::UI::Xaml::Input::PointerRoutedEventArgs& eventArgs)
+    {
+        if (eventArgs.GetCurrentPoint(*this).Properties().IsMiddleButtonPressed())
+        {
+            if (const auto tabViewItem{ sender.try_as<MUX::Controls::TabViewItem>() })

```

Comment:
Checking this again is unnecessary, isn't it? These event handlers are already specific to the `TabViewItem` and won't be called by anything else. I believe the only reason the other methods call `try_as` is to safely get a reference to the `TabViewItem` instance, but here we don't need the instance at all.

Reply:
Sure, removed.